osbenchmark/worker_coordinator/runner.py

```diff
@@ -625,6 +628,35 @@ def __repr__(self, *args, **kwargs):
         return "bulk-index"
 
 
+# TODO: Add retry logic to BulkIndex, so that we can remove BulkVectorDataSet and use BulkIndex.
```

Comment:
Can we just do this instead? Or is it a lot of effort?

Reply:
Yes. It will need an effort since it is going to affect all bulk and we have to add it as not a breaking change.